modules/workergroup/locals.tf

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022, Oracle Corporation and/or affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  tenancy_id = coalesce(var.tenancy_id, var.tenancy_ocid)
+}
+
+locals {
+  cluster = lookup({ # Look for existing cluster by OCID, used in subsequent locals blocks
+    for c in lookup(data.oci_containerengine_clusters.wg_clusters, "clusters", []) : c.id => c if c.id == var.cluster_id
+  }, var.cluster_id, {})
+  compartment_id   = coalesce(var.worker_compartment_id, var.compartment_id, var.compartment_ocid, local.tenancy_id)
+  region           = coalesce(var.region, "us-ashburn-1")
+  mode             = coalesce(var.mode, "node-pool")
+  size             = max(var.size, 0)
+  shape            = coalesce(var.shape, "VM.Standard.E4.Flex")
+  boot_volume_size = coalesce(var.boot_volume_size, 50)
+  memory           = coalesce(var.memory, 8)
+  ocpus            = coalesce(var.ocpus, 1)
+  k8s_version      = "v1.24.1"
+
+  # SSH public key: base64-encoded PEM > raw PEM > file PEM > null
+  ssh_public_key = (
+    var.ssh_public_key != ""
+    ? try(base64decode(var.ssh_public_key), var.ssh_public_key)
+    : var.ssh_public_key_path != "none"
+    ? file(var.ssh_public_key_path)
+  : null)
+
+  # Stable availability domain selection
+  ad_numbers = keys(local.ad_number_to_name)
+  ad_number_to_name = {
+    for ad in data.oci_identity_availability_domains.ad_list.availability_domains :
+    parseint(substr(ad.name, -1, -1), 10) => ad.name
+  }
+
+  # Instance tags - variables + constant
+  defined_tags = merge(coalesce(var.defined_tags, {}), {})
+  freeform_tags = merge(coalesce(var.freeform_tags, {}), {
+    oke_component = "worker",
+  })
+}
+
+locals {
```

Comment:
Again, if there's a reason for them to be in a separate locals block, then please put them in a different file.